hail/python/hail/methods/relatedness/king.py

```diff
@@ -0,0 +1,289 @@
+import hail as hl
+
+from hail.expr.expressions import expr_call
+from hail.expr.expressions import matrix_table_source
+from hail.typecheck import typecheck, nullable
+from hail.utils import deduplicate
+from hail.utils.java import Env
+
+
+@typecheck(call_expr=expr_call, block_size=nullable(int))
+def king(call_expr, *, block_size=None):
+    r"""Compute relatedness estimates between individuals using a KING variant.
+
+    .. include:: ../_templates/req_diploid_gt.rst
+
+    Examples
+    --------
+    Estimate the kinship coefficient for every pair of samples.
+
+    >>> kinship = hl.king(dataset.GT)
+
+    Notes
+    -----
+
+    The following presentation summarizes the methods section of Manichaikul,
+    et. al., but adopts a more consistent notation for matrices.
+
+    Let
+
+    - :math:`i` and :math:`j` be two individuals in the dataset
+
+    - :math:`N^{Aa}_{i}` be the number of heterozygote genotypes for individual
+      :math:`i`.
+
+    - :math:`N^{Aa,Aa}_{i,j}` be the number of variants at which a pair of
+      individuals both have heterozygote genotypes.
+
+    - :math:`N^{AA,aa}_{i,j}` be the number of variants at which a pair of
+      individuals have opposing homozygote genotypes.
+
+    - :math:`S_{i,j}` be the set of single-nucleotide variants for which both
+      individuals :math:`i` and :math:`j` have a non-missing genotype.
+
+    - :math:`X_{i,s}` be the genotype score matrix. Homozygous-reference
+      genotypes are represented as 0, heterozygous genotypes are represented as
+      1, and homozygous-alternate genotypes are represented as
+      2. :math:`X_{i,s}` is calculated by invoking
+      :meth:`CallExpression.n_alt_alleles` on the `call_expr`.
```

Comment:
Is this a matrix or a vector? (Wondering because it refers to a particular sample, which makes me think of one column of a matrix table). What is little `s`?

Reply:
It's a matrix. `s` is for SNP.

Reply:
I added a sentence describing the subscripts.